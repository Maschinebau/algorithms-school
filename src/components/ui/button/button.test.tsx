import { render, fireEvent, screen } from "@testing-library/react"
import { Button } from "./button"

describe("Button component", () => {
  it("calls onClick handler when button is clicked", () => {
    const onClickMock = jest.fn()
    render(<Button text="Click me" onClick={onClickMock} />)
    const button = screen.getByText("Click me")
    fireEvent.click(button)
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })
})

describe("Button Component Snapshot", () => {
  it("renders button with text", () => {
    const { asFragment } = render(<Button text="Click me" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders button without text", () => {
    const { asFragment } = render(<Button />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders disabled button", () => {
    const { asFragment } = render(<Button disabled />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders loader icon when isLoader prop is true", () => {
    const { asFragment } = render(<Button isLoader />)
    expect(asFragment()).toMatchSnapshot()
  })
})
