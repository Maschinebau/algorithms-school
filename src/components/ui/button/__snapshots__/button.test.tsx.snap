// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshot renders button with text 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Click me
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshot renders button without text 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshot renders disabled button 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshot renders loader icon when isLoader prop is true 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;
